index.bs

```diff
@@ -1938,31 +1886,59 @@ Issue: This needs to be generalized to work with realms too.
    </dd>
    <dt>Result Type</dt>
    <dd>
-      <pre class="cddl">
-      EmptyResult
+      <pre class="cddl local-cddl">
+        session.SubscriptionRequestResult = {
+          subscription: session.Subscription,
+        }
       </pre>
    </dd>
 </dl>
 
 <div algorithm="remote end steps for session.subscribe">
 The [=remote end steps=] with |session| and |command parameters| are:
 
-1. Let the |list of event names| be the value of the <code>events</code> field of
-   |command parameters|
+1. Let the |event names| be a [=/set=] containing |command parameters|["<code>events</code>"].
+
+1. Let |newly subscribed navigables| be a [=/set=].
+
+1. Let |top-level traversable context ids| be a [=/set=].
+
+1. Let |input context ids| be a [=/set=] of |command parameters|[<code>contexts</code>].
+
+1. If |input context ids| is not empty:
+
+   1. Let |navigables| be [=get navigables by ids=] given |input context ids|.
+
+   1. If [=list/size=] of |navigables| does not equal [=list/size=] of |input context ids|:
+
+      1. Return [=error=] with [=error code=] [=invalid argument=].
 
-1. Let the |list of navigables| be the value of the <code>contexts</code>
-   field of |command parameters| if it is present or null if it isn't.
+   1. Set |newly subscribed navigables| be [=get top-level traversables=] given |navigables|.
```

Comment:
```suggestion
   1. Set |newly subscribed navigables| to [=get top-level traversables=] given |navigables|.
```

Reply:
Also can we rename `newly subscribed navigables` to something else like `subscription navigables` or something. This confused me into thinking it would end up containing only the navigables that we are _newly_ subscribing to rather than all the (top-level) navigables that the subscription applies to.